### ISSUE TITLE
Oops: Unable to handle kernel NULL pointer dereference at virtual address 0000000000000000

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1083]
```
$ ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- make LLVM=1 -j71 defconfig
$ ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- make LLVM=1 -j71
$ boot-utils/boot-qemu.sh -a arm64 -k . -i
...
[    3.792789] Unable to handle kernel NULL pointer dereference at virtual address 0000000000000000
[    3.793365] Mem abort info:
[    3.793617]   ESR = 0x96000004
[    3.793930]   EC = 0x25: DABT (current EL), IL = 32 bits
[    3.794304]   SET = 0, FnV = 0
[    3.794563]   EA = 0, S1PTW = 0
[    3.794826] Data abort info:
[    3.795079]   ISV = 0, ISS = 0x00000004
[    3.795730]   CM = 0, WnR = 0
[    3.796124] user pgtable: 4k pages, 48-bit VAs, pgdp=000000005d297000
[    3.796562] [0000000000000000] pgd=0000000000000000, p4d=0000000000000000
[    3.797203] Internal error: Oops: 96000004 [#1] PREEMPT SMP
[    3.797691] Modules linked in:
[    3.798256] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 5.8.0-rc4-05952-g9f40e8abeb1b #78
[    3.798995] Hardware name: linux,dummy-virt (DT)
[    3.799536] pstate: 00000005 (nzcv daif -PAN -UAO BTYPE=--)
[    3.799994] pc : pl011_dma_probe+0x138/0x35c
[    3.800341] lr : pl011_dma_probe+0x88/0x35c
[    3.800654] sp : ffff80001000b900
[    3.800951] x29: ffff80001000b9b0 x28: ffff00001d300980 
[    3.801407] x27: ffff00001d850000 x26: ffffbee77c4d2000 
[    3.801837] x25: 0000000000020002 x24: ffff00001d8860f0 
[    3.802278] x23: ffff80001000b968 x22: 0000000000000000 
[    3.802710] x21: 0000000000000000 x20: ffff00001d959c00 
[    3.803136] x19: ffff00001d97ac80 x18: 0000000000000000 
[    3.803570] x17: 0000000000000037 x16: ffffbee77b4cce1c 
[    3.804039] x15: ffffbee77becc34f x14: 000000000000004e 
[    3.804460] x13: ffff0000ffffff00 x12: 0000000000000301 
[    3.804888] x11: 0000000000000005 x10: 0101010101010101 
[    3.805337] x9 : 311cded55353f400 x8 : 311cded55353f400 
[    3.805767] x7 : 0000000000000005 x6 : ffff00001d307237 
[    3.806384] x5 : 0000000000000040 x4 : 0000000000000000 
[    3.806805] x3 : 0000000000000064 x2 : 0000000000000000 
[    3.807233] x1 : ffff00001d850000 x0 : 0000000000000000 
[    3.807775] Call trace:
[    3.808206]  pl011_dma_probe+0x138/0x35c
[    3.808539]  pl011_startup+0x198/0x448
[    3.808860]  uart_port_startup+0x130/0x2e0
[    3.809171]  uart_port_activate+0x5c/0xcc
[    3.809466]  tty_port_open+0x94/0x25c
[    3.809781]  uart_open+0x20/0x30
[    3.810076]  tty_open+0x11c/0x248
[    3.810374]  chrdev_open+0x19c/0x200
[    3.810681]  do_dentry_open+0x22c/0x388
[    3.810994]  vfs_open+0x2c/0x38
[    3.811276]  path_openat+0x840/0xbb8
[    3.811577]  do_filp_open+0x78/0x124
[    3.811904]  do_sys_openat2+0x80/0x154
[    3.812214]  do_sys_open+0x5c/0x84
[    3.812512]  console_on_rootfs+0x24/0x54
[    3.812839]  kernel_init_freeable+0xf8/0x144
[    3.813175]  kernel_init+0x14/0x18c
[    3.813470]  ret_from_fork+0x10/0x30
[    3.814085] Code: 90006501 91162821 9401481b 17ffffdb (f94002a8) 
[    3.814958] ---[ end trace 5e084ff633b1f1d3 ]---
[    3.815878] Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
[    3.816650] Kernel Offset: 0x3ee76b000000 from 0xffff800010000000
[    3.817065] PHYS_OFFSET: 0xffffe5d180000000
[    3.817535] CPU features: 0x7e0152,20c02020
[    3.817873] Memory Limit: none
[    3.818349] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b ]---

**Comment by nathanchance** [comment]
From https://github.com/ClangBuiltLinux/continuous-integration/pull/286:

> Note, as of next-20200707, arm32 and arm64 boot is broken but should be
fixed in next-20200708 for reasons unrelated to LLVM:
> 
> https://lore.kernel.org/dmaengine/5f036d83.1c69fb81.10199.06d0@mx.google.com/
> https://lore.kernel.org/dmaengine/159404871194.45151.3076873396834992441.stgit@djiang5-desk3.ch.intel.com/

**Comment by nathanchance** [comment]
Fixed in next-20200708.
